### DISCUSSION TITLE
How to distinguish mouse scrolling and scrollTo api scrolling ?

### DISCUSSION THREAD

**Discussion by NexxLuo** [petyosi/react-virtuoso#435] [Q&A]
I have two vitual list ,they  need to scroll synchronously .  I use then `scrollTo` change the list scroll position , but they will scroll in a loop ? In the another library `react-window` ,  `onScroll`  event has an parameter called `scrollUpdateWasRequested`.
How can i do with `react-virtuoso` ?

**Comment by petyosi** [comment]
This is not supported currently, it does sound like a nice to have enhancement. As an alternative, You can maintain `scrollUpdateFlagWasRequested` as an external flag in your own state. Feel free to post an example, it can be useful for others as well.

**Comment by farhadibehnamdev** [comment]
I added an `id` to the Virtuoso component. In useEffect, a `wheel ` event is added to the Virtuoso component each time a user scrolls the mouse so that it can be detected.

```
  useEffect(() => {
    const vrtElement = document.getElementById("vrt");
    vrtElement?.addEventListener("wheel", doSemething);

    return () => {
      vrtElement?.removeEventListener("wheel", doSemething);
    };
  }, []);
```




```
<Virtuoso
      ref={virtuRef}
      style={{
        height: "100vh",
      }}
      id="vrt"
      data={getDataEntries()}
      itemContent={(index, data) => {return ()}
       />
```

**Reply by eslintinit** [reply]
saved my day thx!
